hr_public_holidays/models/hr_public_holidays.py

```diff
@@ -33,11 +35,55 @@ class HrPublicHolidays(models.Model):
         'Country'
     )
 
+    @api.model
+    def create(self, values):
+        holiday = super(HrPublicHolidays, self).create(values)
+        for line in holiday.line_ids:
+            leave = self.env['resource.calendar.leaves'].create(
```

Comment:
Instead of creating `resource.calendar.leaves` directly in the create() or the write() methods, I would prefer a solution similar to the one we adopted [here](https://github.com/onesteinbv/addons-onestein/blob/10.0/hr_public_holidays_leaves/models/hr_public_holidays.py).

The idea is to create a [Status field](https://github.com/onesteinbv/addons-onestein/blob/10.0/hr_public_holidays_leaves/models/hr_public_holidays.py#L15) in the `hr.holidays.public` model and adopt a minimalistic workflow [`draft`->`validate`](https://github.com/onesteinbv/addons-onestein/blob/10.0/hr_public_holidays_leaves/views/hr_public_holiday.xml#L10-L13).